docs/07-lifecycle.md

```diff
@@ -2,8 +2,68 @@
 ## Development
 ## Submit to Marketplace
 ## Provision
+### App Bundle
+Bundle is a convenient way to deliver an app to the Mattermost ecosystem. It provides a way to organize code and resources needed for an app to run. A bundle is created by the developer of the app. Mattermost uses bundles to provision and install/uninstall apps. 
+App bundle contains `manifest.json` file, `static/` folder and one or several lambda function bundles.
+- `static/` folder contains all the static files app needs. For the Mattermost AWS Apps static files are automatically provisioned and stored in the dedicated AWS S3 bucket(`mattermost-apps-bucket-production`). Apps can use them whenever they wish by providing the static file name to the Proxy Plugin. For the third-party hosted AWS apps static files are stored in the different S3 bucket(specified by the third-party). For the HTTP Apps, when creating a server, developer should store the static files in the `/static/file_name` relative URL.
+- `manifest.json` file contains details about the app such as appID, appVersion, appType(http or an AWS app), requested permissions, requested locations, and information about the  functions such as function Path, name, runtime and handler.
+- Each of the lambda function bundles is a valid and runnable AWS lambda function, which are provisioned in the AWS by the [Mattermost Apps Cloud Deployer](https://github.com/mattermost/mattermost-apps-cloud-deployer).
+AWS Lambda function bundle is a `.zip` file which contains scripts or compiled programs and their dependencies. Note that it must be less than 50 MB. Exact specification of the bundle varies for different runtimes. For example one can see more details for `node.js` bundles [here](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-package.html) 
+
+### Provisioning In The 3rd Party AWS
+Provisioning in the third party AWS cloud environment is done by the **appsctl** tool using the command
+
+`go run ./cmd/appsctl/ provision app /PATH/TO/YOUR/APP/BUNDLE`
```

Comment:
Should we recommend `go install` instead? Otherwise we also have to document the `git clone`.

Reply:
I was under impression that app developer would already have the proxy plugin cloned. Isn't it a case?

Reply:
I don't think we mention that anywhere. Encouraging people to run `go install github.com/mattermost/mattermost-plugin-apps/cmd/appsctl@latest` seems like a good way to get a user starting.